api/src/main/java/org/openmrs/module/reporting/config/ReportLoader.java

```
@@ -0,0 +1,251 @@
+                    parameters.add(new Parameter(parameterDescriptor.getKey(), parameterDescriptor.getLabel(), Class.forName(parameterDescriptor.getType())));
+                } catch (Exception e) {
+                    log.error("Unable to configure parameter " + parameterDescriptor.getKey(), e);
+                }
+            }
+        }
+        return parameters;
+    }
+
+
+    public static Map<String, Object> constructMappings(List<Parameter> parameters) {
+        // TODO: Mike is the correct and/or necessary?
+        Map<String,Object> mappings = new HashMap<String, Object>();
+        for (Parameter parameter : parameters) {
+            mappings.put(parameter.getName(), "${" + parameter.getName() + "}");
+        }
+        return mappings;
+    }
+
+    public static DataSetDefinition constructDataSetDefinition(DataSetDescriptor dataSetDescriptor, List<Parameter> parameters) {
+
+        DataSetDefinition dsd = null;
+
+        if ("sql".equalsIgnoreCase(dataSetDescriptor.getType())) {
+            dsd = constructSQLFileDataSetDefinition(dataSetDescriptor);
+        }
+        else {
+            log.error("Unsupported data set descriptor type: " + dataSetDescriptor.getType());
+        }
+
+        if (parameters != null) {
+            for (Parameter parameter : parameters) {
+                dsd.addParameter(parameter);
+            }
+        }
+
+        return dsd;
+    }
+
+    public static SqlFileDataSetDefinition constructSQLFileDataSetDefinition(DataSetDescriptor dataSetDescriptor) {
+
+        SqlFileDataSetDefinition dsd = new SqlFileDataSetDefinition();
+
+        File sqlFile = new File(getReportingDescriptorsConfigurationDir() + File.separator + dataSetDescriptor.getConfig());
+
+        if (sqlFile.exists()) {
+            dsd.setSqlFile(sqlFile.getAbsolutePath());
+        }
+        else {
+            log.error("SQL file " + dataSetDescriptor.getConfig() + " not found");
```

Same here, why not fail hard?  This is much less likely to get noticed, but is evidence of an unintended configuration error.  Just fail with a runtime exception.